src/hardware/bk-1856d/api.c

```diff
@@ -0,0 +1,228 @@
+/*
+ * This file is part of the libsigrok project.
+ *
+ * Copyright (C) 2020 LUMERIIX/danselmi <.>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <config.h>
+#include "protocol.h"
+
+#define SERIALCOMM "9600/8n1/dtr=1/rts=0"
+
+static const uint32_t scanopts[] = {
+	SR_CONF_CONN,
+};
+
+static const uint32_t drvopts[] = {
+	SR_CONF_MULTIMETER,
+};
+
+static const uint32_t devopts[] = {
+	SR_CONF_CONTINUOUS,
+	SR_CONF_TIMEBASE | SR_CONF_SET |SR_CONF_GET| SR_CONF_LIST, // used to select gate time
+	SR_CONF_LIMIT_SAMPLES | SR_CONF_SET,
```

Comment:
You could add `SR_CONF_LIMIT_MSEC` here and in the getter and setter functions now.

Reply:
My bad. I wrote this before seeing your last comment.